registration/include/pcl/registration/icp.h

```diff
@@ -154,6 +154,11 @@ namespace pcl
         convergence_criteria_.reset(new pcl::registration::DefaultConvergenceCriteria<Scalar> (nr_iterations_, transformation_, *correspondences_));
       };
 
+      IterativeClosestPoint(const IterativeClosestPoint&) = delete;
```

Comment:
Can we add comment here specifying why we're disallowing copy and move. 

Because DefaultConvergenceCriteria is holding references to members of its enclosing container object.